### DISCUSSION TITLE
Logging questions

### DISCUSSION THREAD

**Discussion by Cantello** [creativeprojects/resticprofile#120] [Q&A] [answered]
Potential newbie questions here:

- When backing up an Ubuntu Server host, I have a logfile written with `schedule-log: /var/log/resticprofile-backup.log`. However, (only a minor annoyance), it contains special characters, probably escape codes from the terminal output. Any idea how to suppress them in the logfile? `--no-ansi` is already present in the systemd timer.
- In the logfile, there are no time stamps, can these be activated somehow?
- Should/could I rely on ` run-after-fail: curl -fsS --retry 3 --data-raw \"$ERROR_STDERR\" https://hc-ping.com/` to notify me of errors or is the logfile more reliable?

**Comment by creativeprojects** [comment, marked as answer]
Sorry for the late answer: I got the notification, then I forgot about it 😛 

> When backing up an Ubuntu Server host, I have a logfile written with schedule-log: /var/log/resticprofile-backup.log. However, (only a minor annoyance), it contains special characters, probably escape codes from the terminal output. Any idea how to suppress them in the logfile? --no-ansi is already present in the systemd timer.

Do they seem to be coming from the resticprofile log lines, or from restic? `--no-ansi` should have worked but maybe we introduced a bug recently.

> In the logfile, there are no time stamps, can these be activated somehow?

I'm surprised these are not there, I'll need to have a look why they disappeared

> Should/could I rely on run-after-fail: curl -fsS --retry 3 --data-raw \"$ERROR_STDERR\" https://hc-ping.com/ to notify me of errors or is the logfile more reliable?

They're both equally reliable. In the next version there will be an even better way: #114

**Reply by Cantello** [reply]
Late answere here as well, but now I have been using resticprofile and I am absolutely *loving* it!

With regards to the non-ANSI characters, it was even better than them coming from restic: I put in some before-actions to power down my docker containers prior to backing them up and it was docker's log that appeared as special characters in the log. The timestamps were also present, I failed to see that that they were only printed when resticprofile did some actions.

As for the http hooks, they way they are working now is good enough for me, so far the only time they triggered was due to me botching up the config file.

Looking forward to future versions! I even tried kopia as the main 'contender' for CLI-based multiplatform backup and the sole reason I did not switch was resticprofile... :-)
